provider/pdns_test.go

```diff
@@ -222,6 +223,34 @@ var (
 		Rrsets: []pgo.RrSet{RRSetCNAMERecord, RRSetTXTRecord, RRSetMultipleRecords},
 	}
 
+	ZoneList = []pgo.Zone{
+		ZoneEmpty,
+		ZoneEmpty2,
+	}
+
+	PartitionResultFilteredEmptyFilter = []pgo.Zone{
```

Comment:
Can you move all those declarations inside the test? It looks like you've been consistent with the style of this file, but I don't see the point in having global variables for something that is used in a single test. It's much harder to read in my opinion and we should not abuse global things when we can keep them local to the examined function.

Reply:
I've moved the Partition function specific ones inside the function. However I have left a few global ones like the `DomainFilterList` and the `DomainFilterClient` ones as they can be used in other tests. Hope that is okay.